synology_dsm/api/core/upgrade.py

```diff
@@ -0,0 +1,26 @@
+# -*- coding: utf-8 -*-
+"""DSM Upgrade data and actions."""
+
+
+class SynoCoreUpgrade(object):
+    """Class containing upgrade data and actions."""
+
+    API_BASE_KEY = "SYNO.Core.Upgrade"
+    API_SERVER_KEY = API_BASE_KEY + ".Server"
+
+    def __init__(self, dsm):
+        self._dsm = dsm
+        self._data = {}
+
+    def update(self):
+        """Updates Upgrade data."""
+        raw_data = self._dsm.get(self.API_SERVER_KEY, "check")
+        if raw_data:
+            self._data.update(raw_data["data"])
+
+    @property
+    def update_available(self):

```

Comment:
Any other interesting property ?

Reply:
there are no other properties, but actions available (like perform a update). But because of missing an available update of my diskstation, I can not test it and therefore did not implement it for now. But I will do until an update for my diskstation is availabl.

Reply:
I'm good with that (maybe it will have a link to DSM release notes).

On following up properties, we should not forget to handle `if not update_available: return None`.